lib/model/item_group.dart

```diff
@@ -0,0 +1,16 @@
+import 'package:lists/model/item.dart';
+import 'package:lists/model/list_model_item_group.dart';
+
+/// ItemGroup:
+///   - an abstract base class for a group of items with a title.
+abstract class ItemGroup {
+  String? get title;
+  set title(String? newTitle);
+
+  Iterable<Item> itemsView();
+  int get itemCount;
+
+  const ItemGroup();
+
+  Future<ListModelItemGroup> asListModelItemGroup();
```

Comment:
```suggestion
  ItemGroup get originalGroup;
```

Reply:
I think the idea behind this was to be able to get the "original" item group from a filtered item group?

Reply:
Yes, the issue I was running into was due to the fact that in `ListWidget`, I store `groupsToBeDisplayed` as a `List` of `AbstractItemGroup` (so that I can display the groups initially gotten from the `listModel.groupsView()` or groups that have been filtered). When one of these groups is edited (say, the title is changed), the changes need to be put to isar. However, only the item groups stored in list models (and therefore the database) can be put, so I needed a way of getting an item group that can be put to the database from any abstract item group.

I can look into improving my design.

Reply:
In a normal database design, you would just store the ID of the original group, then use that ID to get the group that should be updated. You can redo your design, or just follow the renaming suggestions I gave you. I don't think your design is that bad, it just needs a few small improvements.